frontend/src/metabase/redux/uploads.ts

```diff
@@ -44,64 +41,83 @@ export const hasActiveUploads = (state: State) =>
 
 export const uploadFile = createThunkAction(
   UPLOAD_FILE_TO_COLLECTION,
-  (file: File, collectionId: CollectionId) => async (dispatch: Dispatch) => {
-    const id = Date.now();
-
-    const clear = () =>
-      setTimeout(() => {
-        dispatch(clearUpload({ id }));
-      }, CLEAR_AFTER_MS);
-
-    dispatch(
-      uploadStart({
-        id,
-        name: file.name,
-        collectionId,
-      }),
-    );
-
-    if (file.size > MAX_UPLOAD_SIZE) {
-      dispatch(
-        uploadError({
-          id,
-          message: t`You cannot upload files larger than ${MAX_UPLOAD_STRING} MB`,
-        }),
-      );
-      clear();
-      return;
-    }
-
-    try {
-      const formData = new FormData();
-      formData.append("file", file);
-      formData.append("collection_id", String(collectionId));
-      const response = await CardApi.uploadCSV(formData);
+  ({
+      file,
+      collectionId,
+      tableId,
+    }: {
+      file: File;
+      collectionId?: CollectionId;
+      tableId?: TableId;
+    }) =>
+    async (dispatch: Dispatch) => {
+      const id = Date.now();
+
+      const clear = () =>
+        setTimeout(() => {
+          dispatch(clearUpload({ id }));
+        }, CLEAR_AFTER_MS);
 
       dispatch(
-        uploadEnd({
+        uploadStart({
           id,
-          modelId: response,
+          name: file.name,
+          collectionId,
+          tableId,
         }),
       );
 
-      dispatch(Collections.actions.invalidateLists());
-      clear();
-    } catch (err: any) {
-      dispatch(
-        uploadError({
-          id,
-          message: t`There was an error uploading the file`,
-          error: err?.data?.message ?? err?.data,
-        }),
-      );
-    }
-  },
+      if (file.size > MAX_UPLOAD_SIZE) {
+        dispatch(
+          uploadError({
+            id,
+            message: t`You cannot upload files larger than ${MAX_UPLOAD_STRING} MB`,
+          }),
+        );
+        clear();
+        return;
+      }
+
+      try {
+        const formData = new FormData();
+        formData.append("file", file);
+        formData.append("collection_id", String(collectionId));
+
+        const response = await (tableId
```

Comment:
`tableId` can't be `0` or `''`, right?

I ask about `''` because I see that a `TableId` can be a number or a string.

Reply:
sometimes 	`tableId` can be a reference to another question when it's a query that queries another question like `card__1234`